### PR TITLE
fix(router-plugin): add the rspack/rsbuild handler for file-based routing

### PR DESCRIPTION
**Pull request by SeanCassiere**

### CODE REVIEW

File: packages/router-plugin/src/composed.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+> = (options = {}, meta) => {
+  const generatorResult = unpluginRouterGeneratorFactory(options, meta)
+
+  const generator = Array.isArray(generatorResult)
+    ? generatorResult
+    : [generatorResult]
+
+  const codeSplitterResult = unpluginRouterCodeSplitterFactory(options, meta)
+  const codeSplitter =
+    meta.framework === 'rspack'
```
Comment (schiller-manuel) [inline packages/router-plugin/src/composed.ts:18]: why does rspack behave differently here?

Comment (SeanCassiere) [reply packages/router-plugin/src/composed.ts:18]: Yeah, it doesn't support the `resolveId` and `transform` calls.

Need to do more investigation into whether I can get this same functionality using their compiler hooks. If not, I'll add an asterisk next to experimental code-splitting not working.

File: docs/framework/react/guide/code-splitting.md [hunk 2]
```diff
@@ -180,6 +180,8 @@ Tada! 🎉
 
 > ⚠️ **Experimental:** This feature is experimental and currently in development. It is not recommended for production use.
 
+> ⚠️ **Important:** This feature is currently does not work with Rspack/Rsbuild. If you are familiar with [Unplugin](https://unplugin.unjs.io) or Rspack, please let us know if you can help us get this working!
```
Comment (chenjiahan) [inline docs/framework/react/guide/code-splitting.md:183]: Glad to see that TanStack Router now supports Rspack.

For th unplugin issue mentioned here, it seems to be caused by Rspack not supporting the unplugin resolveId hook. I have reported this to the Rspack team and expect to see support for it in the future.

**Comment by zngly-vlad** [comment]
@SeanCassiere @chenjiahan 

A heads up - It seems that unplugin will fully support rspack with this pr https://github.com/unjs/unplugin/pull/388

**Comment by chenjiahan** [comment]
Yes, this PR is sent by our team member and unplugi now fully support Rspack.

**Comment by SeanCassiere** [comment]
@zngly-vlad @chenjiahan Thank you!

The router-plugin's rspack export should currently have full compat with the Vite and Webpack plugins.

**Pull request merged**